libappstream-glib/as-store.c

```diff
@@ -1830,31 +1830,46 @@ as_store_from_root (AsStore *store,
 }
 
 static gboolean
-as_store_load_yaml_file (AsStore *store,
-			 GFile *file,
-			 AsAppScope scope,
-			 GCancellable *cancellable,
-			 GError **error)
+is_dep11_data (GBytes *bytes)
+{
+	const gchar *data;
+	gsize i, size, line_start;
+
+	data = g_bytes_get_data (bytes, &size);
+	line_start = 0;
+	for (i = 0; i < 1000 && i < size; i++) {
+		gsize line_length;
+		if (data[i] != '\n')
+			continue;
+
+		/* look for DEP-11 header */
+		line_length = i - line_start;
+		if (strncmp (data + line_start, "File: DEP-11", line_length) == 0)
```

Comment:
this seems a bit clumsy -- can't we just do something like `g_strstr_len (data, "File: DEP-11", size) != NULL`

Reply:
I thought that would be not a particularly strong indication, but if it's not a DEP-11 file then it's going to fail in `as_store_load_yaml_data` so that seems reasonable.